static/app/components/smartSearchBar/index.tsx

```diff
@@ -749,13 +769,222 @@ class SmartSearchBar extends React.Component<Props, State> {
+  }
+
+  getValidOps(filterToken: TokenResult<Token.Filter>): readonly TermOperator[] {
+    const types = filterToken.invalid?.expectedType ?? [filterToken.filter];
+    const filterType = interchangeableFilterOperators[types[0]]
+      ? interchangeableFilterOperators[types[0]][0]
+      : types[0];
+    const config = filterTypeConfig[filterType];
+    return config.validOps ?? [''];
+  }
+
+  generateOpAutocompleteGroup(
+    validOps: readonly TermOperator[],
+    tagName: string
+  ): AutocompleteGroup {
+    const operatorItems = this.generateOpAutocompleteEntries(validOps, tagName);
+    return {
+      searchItems: operatorItems,
+      recentSearchItems: undefined,
+      tagName: '',
+      type: 'tag-operator' as ItemType,
+    };
+  }
+
+  withinTokenLocation(node: {location: LocationRange}, position: number) {
+    return position >= node.location.start.offset && position <= node.location.end.offset;
+  }
+
+  getCursorToken = (AST: ParseResult, cursor: number): ParseResult[number] | null => {
+    // traverse AST to find first matching filter based on cursor position
+    for (const node of AST) {
+      if (node !== Token.Spaces && this.withinTokenLocation(node, cursor)) {
+        // traverse into a logic group to find specific filter
+        if (node.type === Token.LogicGroup) {
+          return this.getCursorToken(node.inner, cursor);
+        }
+        return node;
+      }
+    }
+    return null;
+  };
+
   updateAutoCompleteItems = async () => {
     if (this.blurTimeout) {
       clearTimeout(this.blurTimeout);
       this.blurTimeout = undefined;
     }
 
     const cursor = this.getCursorPosition();
+    const {organization} = this.props;
+    const AST = this.state.parsedQuery;
+    if (AST && organization.features.includes('search-syntax-highlight')) {
+      const filterNode = this.getCursorToken(AST, cursor);
+      if (filterNode && filterNode.type === Token.Filter) {
+        // check if we are on the tag, value, or operator
+        if (this.withinTokenLocation(filterNode.value, cursor)) {
+          const node = filterNode.value;
+          const tagName = filterNode.key.text;
+
+          const supportedTags = this.props.supportedTags ?? {};
+          const tag = supportedTags[tagName];
+
+          const valueGroup = await this.generateValueAutocompleteGroup(
+            tagName,
+            node.text
+          );
+          const autocompleteGroups = valueGroup ? [valueGroup] : [];
+
+          // show operator group if at beginning of value
+          if (cursor === node.location.start.offset) {
+            const opGroup = this.generateOpAutocompleteGroup(
+              this.getValidOps(filterNode),
+              tag.key
+            );
+            autocompleteGroups.unshift(opGroup);
+          }
+
+          this.updateAutoCompleteStateMultiHeader(autocompleteGroups);
+          return;
+        } else if (this.withinTokenLocation(filterNode.key, cursor)) {
```

Comment:
You don't need the `else` here. You're already returning at the end of the `if` statement just before this.

I would add a space between these two cases as well


```tsx
  ....
  this.updateAutoCompleteStateMultiHeader(autocompleteGroups);
  return;
}

if (this.withinTokenLocation(filterNode.key, cursor)) {
  const node = filterNode.key;
  ....
```